packages/chatbot-server-mongodb-public/src/tracing.ts

```diff
@@ -0,0 +1,84 @@
+import { AppConfig } from "mongodb-chatbot-server";
+import { SomeMessage, UserMessage } from "mongodb-rag-core";
+import { llmDoesNotKnowMessage } from "./systemPrompt";
+
+export const makeAddMessageToConversationUpdateTrace: (
+  k: number
+) => AppConfig["conversationsRouterConfig"]["addMessageToConversationUpdateTrace"] = (
+  k
+) =>
+  async function ({ traceId, addedMessages, logger }) {
+    const tracingData = extractTracingData(addedMessages);
+    logger.updateSpan({
+      id: traceId,
+      tags: tracingData.tags,
+      scores: {
+        RejectedQuery: tracingData.rejectQuery === true ? 1 : null,
```

Comment:
Curious - is there a meaningful difference between a score of `null` and a score of `0`?

Reply:
yep, `null` means 'no score registered', whereas 0 is the lowest possible score